contributors/devel/testing.md

```diff
@@ -225,3 +225,40 @@ version and the watch cache test is skipped.
 ## End-to-End tests
 
 Please refer to [End-to-End Testing in Kubernetes](e2e-tests.md).
+
+## Running your contribution in the Kubernetes CI
+Once you open a PR, prow will run pre-submit tests in CI.
+
+### Troubleshooting a failure
+Click on `Details` and look at the [`gubernator`](gubernator.k8s.io/) output for the test.
+
+#### Troubleshooting failures/flakes that are not caused by your change
```

Comment:
possibly unify the two sections?

we can explain that if the gubernator output seems unrelated to the change they might try to first call `/retest` them self. but if a test continues to fail, to contact a reviewer/maintainer.

usually what happens is that contributors first ping in the PR and if nobody responds they go asking on slack.

Reply:
Good point, we unified them under a single title and structured them a bit more.